.github/workflows/jira-sync.yml

```diff
@@ -0,0 +1,118 @@
+name: Sync GitHub Issues to JIRA
+on:
+  workflow_call: 
+    inputs:
+      jira-url:
+        required: true
+        type: string
+        description: |
+          Specifies the Base URL of the JIRA instance to sync against.
+      jira-project:
+        required: true
+        type: string
+        description: |
+          Specifies the JIRA project key for the JIRA project into which issues should be sync'd.
+      cross-links-file:
+        required: false
+        type: string
+        default: cross-links.yaml
+        description: |
+          Specifies a file that will be used to store the cross-links between GitHub and JIRA to ensure that issues are 
+          not unnecessarily sync'd multiple times.
+      issue-mapping-file:
+        required: true
+        type: string
+        description: |
+          Specifies the file that contains the mapping rules controlling how GitHub Issues are mapped into JIRA Issue 
+          Types, this allows ensuring that the correct issue types are created on the JIRA side during sync.
+      extra-sync-options:
+        required: false
+        type: string
+        description: |
+          Specifies additional options to pass into the jira-sync command when sync'ing issues e.g. `--skip-existing` 
+          if you only want to sync new issues.
+
+          Note that these options are added to the start of the command options so they cannot override the options that
+          this action will automatically set from your other inputs and secrets.
+    secrets:
+      JIRA_USERNAME:
+        required: true
+        description: |
+          A JIRA Username (usually an email address) which can be used to authenticate against the 
+          configured JIRA instance in conjunction with the JIRA_TOKEN secret.
+      JIRA_TOKEN:
+        required: true
+        description: |
+          A JIRA Token (created by a specific user account) which can be used to authenticate against the
+          configured JIRA instance in conjunction with the JIRA_USERNAME secret.
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: read
+    concurrency:
+      group: jira-sync
+      cancel-in-progress: false
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Java 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Install jira-sync
+        uses: jaxxstorm/action-install-gh-release@v1.11.0
+        with: # Grab the latest version
```

Comment:
Could be pinned to a specific release if we wished?

We control the telicent-oss/jira-sync repository so only we can create new releases in it

Reply:
I'm a little ignorant to github actions specifics here. My assumption is that the v1.11.0 tag of the third party action could be moved to newer code and we would continue to load it into our workflow without verification. 

By fixing the version to a commit hash instead this would not be possible.

> Pin actions to a full length commit SHA
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
Latest version of telicent-oss/jira-sync is fine. Pinning the third party action that is doing the install is advisable.